source/Camera/Basic/C++Halcon/HalconConnectUsingZivid/HalconConnectUsingZivid.cpp

```diff
@@ -0,0 +1,97 @@
+#include <Zivid/Zivid.h>
+#include <halconcpp/HalconCpp.h>
+#include <opencv2/core/core.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+
+#include <chrono>
+#include <iostream>
+
+
+void savePointCloud(const HalconCpp::HObjectModel3D &model, const std::string &filename)
+{
+	model.WriteObjectModel3d(HalconCpp::HString{ "ply" },
+		HalconCpp::HString{ filename.c_str() },
+		HalconCpp::HString{ "invert_normals" },
+		HalconCpp::HString{ "false" });
+}
+
+
+HalconCpp::HObjectModel3D zividToHalconPointCloud(const Zivid::PointCloud &pointCloud)
+{
+	// Extract data from Zivid point cloud
+	const auto width = static_cast<long int>(pointCloud.width());
+	const auto height = static_cast<long int>(pointCloud.height());
+
+	std::vector<float> array1dX(width * height, 0.0);
+	std::vector<float> array1dY(width * height, 0.0);
+	std::vector<float> array1dZ(width * height, 0.0);
+	std::vector<float> array1dR(width * height, 0.0);
+	std::vector<float> array1dG(width * height, 0.0);
+	std::vector<float> array1dB(width * height, 0.0);
+
+	auto data = pointCloud.dataPtr();
+	size_t idx = 0;
+	for (size_t i = 0; i < width; ++i)
+		for (size_t j = 0; j < height; ++j)
+		{
+			idx = width * j + i;
+			if (!isnan(data[idx].x))
+			{
+				array1dX[idx] = data[idx].x;
+				array1dY[idx] = data[idx].y;
+				array1dZ[idx] = data[idx].z;
+				array1dR[idx] = data[idx].red();
+				array1dG[idx] = data[idx].green();
+				array1dB[idx] = data[idx].blue();
+			}
+		}
+
+	std::cout << "Constructing ObjectModel3D based on x,y and z data" << std::endl;
+	const auto X = HalconCpp::HImage{ "real", width, height, array1dX.data() };
+	const auto Y = HalconCpp::HImage{ "real", width, height, array1dY.data() };
+	const auto Z = HalconCpp::HImage{ "real", width, height, array1dZ.data() };
+	HalconCpp::HObjectModel3D objectModel3D(X, Y, Z);
+
+	HalconCpp::SetObjectModel3dAttribMod(objectModel3D, "red", "points", HalconCpp::HTuple{ array1dR.data(), width * height });
```

Comment:
std::cout << "Adding RGB to ObjectModel3D" << std::endl;

Reply:
done